godot-cell/src/cell.rs

```diff
@@ -88,34 +89,42 @@ pub(crate) struct GdCellInner<T> {
 impl<T> GdCellInner<T> {
     /// Creates a new cell storing `value`.
     pub fn new(value: T) -> Pin<Box<Self>> {
-        let cell = Box::pin(Self {
-            state: Mutex::new(CellState::new()),
-            value: UnsafeCell::new(value),
-            _pin: PhantomPinned,
-        });
+        let mut uninitialized_cell: Box<MaybeUninit<Self>> = Box::new_uninit();
 
-        cell.state.lock().unwrap().initialize_ptr(&cell.value);
+        unsafe {
+            let value_ptr = &raw mut (*uninitialized_cell.as_mut_ptr()).value;
+            // SAFETY: Box::pin(...) is equivalent to Box::into_pin(Box::...) therefore `value_ptr`
+            // is and will stay valid (i.e. will refer to the same place after pinning).
+            value_ptr.write(UnsafeCell::new(value));
+            (&raw mut (*uninitialized_cell.as_mut_ptr()).state)
+                .write(UnsafeCell::new(CellState::new(value_ptr.as_ref().unwrap())));
 
-        cell
+            Box::into_pin(uninitialized_cell.assume_init())
+        }
```

Comment:
Please split into individual `unsafe` blocks and extract all safe operations outside. It will be more verbose, but with a per-unsafe `// SAFETY` statement also more clear.

For example, `as_mut_ptr()`, `UnsafeCell::new()`, `Box::into_pin()` etc. are all safe.